pages/blog/the-reference-rabbit-hole.md

```diff
@@ -0,0 +1,275 @@
+```
+and `test.json` is defined as:
+
+```json
+{"minLength": 5, "format": "email"}
+```
+
+JSON Schema draft 2019-09, assumes that the references are resolved similar to:
+```json
+{"$ref": {"minLength": 5, "format": "email"}, "minLength": 7, "maxLength": 12}
+```
+
+This is because in validation, you want to validate that the input data is valid against the referenced schema and should [**not** be seen as a kind of merging behavior](https://github.com/APIDevTools/json-schema-ref-parser/issues/145):
+```json
+{"format": "email", "minLength": 7, "maxLength": 12}
+```
+
+This behavior is different from what is assumed when using AsyncAPI, as the last option, is more aligned with expected behavior.
+
+Furthermore, now, each schema can define it's own `$schema` that they follow, instead of ONLY being available at the root... 
+
+This leaves the question, how can we make sure that we stay consistent and don't introduce more confusion into the AsyncAPI specification? This difference is what triggered the last issue in the [spec 649](https://github.com/asyncapi/spec/issues/649).
+
+## Hard to find tooling
+This leaves us with one huge deficit, that there are so many different behaviors for references that tooling mix and matches between the specifications and what they solve.
+
+One of the most used tooling for dereferencing stuff in JS, and the one we are using is from [APIDevTools called json-schema-ref-parser](https://github.com/APIDevTools/json-schema-ref-parser). We actually use this tool to ensure **ANY** encounters of `$ref` are dereferenced, so the tool has direct access to the schema, without it having to look elsewhere for it. 
+
+However, the tool starting out being build **ONLY** for dereferencing `$ref` based on the [JSON Reference specification and the JSON Pointer specification](https://github.com/APIDevTools/json-schema-ref-parser/issues/22#issuecomment-231783185).  At least it was, now it's not easy to figure it out what it is for, as [it allows extra properties](https://github.com/APIDevTools/json-schema-ref-parser/issues/232) but [$id is not taken into account](https://github.com/APIDevTools/json-schema-ref-parser/issues/136).
+
+This leaves us in a big of a struggle, as [there are not many alternatives](https://json-schema.org/implementations.html#general-processing), for JS [@hyperjump/json-schema-core](https://github.com/jdesrosiers/json-schema-core) looks promising, but there are no tooling that our [Go parser](https://github.com/asyncapi/parser-go) can use.
+
+And with no official or community tooling, we are left with having to develop it our selves to adopt the spec...
+
+# Final word
+That concludes the rabbit hole me and Sergio went down, for a simple `$ref` keyword... (ONE KEYWORD! :sweat_smile:)
+
+All I can add here, in the end, is a serious :pray: that we start switching the mindset from tooling for specification is something others build, to something we all have to build together. 
```

Comment:
```suggestion
All I can add here, in the end, is a serious :pray: that we start switching the mindset from tooling for specification others build, to something we all have to build together. 
```

Reply:
I find this suggestion hard to read 🧐 Don't you?

Reply:
The original sentence does not have sense to me. At least that `is` before `something`.

Reply:
Both the original sentence and Sergio's read awkwardly in English TBH, but no worries because I will fix it :) 

I might add my review as a separate commit in most likelihood, if I find myself wanting to make too many editorial suggestions. Stay tuned :D

Reply:
Continue https://github.com/asyncapi/website/pull/461/files#r753587410